File: examples/src/Shapes/GetLightRigEffectiveData.cs
import aspose.slides as slides
using Aspose.slides.Examples.CSharp
using System
using System.Collections.Generic
using System.Linq
using System.text
using System.Threading.Tasks

namespace CSharp.shapes
{
    class GetLightRigEffectiveData
    {
        public static void Run()
        {

            #ExStart:GetLightRigEffectiveData

            # The path to the documents directory.
            dataDir = RunExamples.GetDataDir_Shapes()

            using (Presentation pres = new Presentation(dataDir + "Presentation1.pptx"))
            {
                IThreeDFormatEffectiveData threeDEffectiveData = pres.slides[0].shapes[0].three_dformat.get_effective()

                print("= Effective light rig properties =")
                print("Type: " + threeDEffectiveData.light_rig.light_type)
                print("Direction: " + threeDEffectiveData.light_rig.direction)


            }

            #ExEnd:GetLightRigEffectiveData
        }
    }
}

File: examples/src/Shapes/GetLightRigEffectiveData.py
import aspose.slides as slides
            
#ExStart:GetLightRigEffectiveData

# The path to the documents directory.
dataDir = "./examples/data/"
outDir = "./examples/out/"

with slides.Presentation(dataDir + "shapes_3d_effective.pptx") as pres:
    threeDEffectiveData = pres.slides[0].shapes[0].three_dformat.get_effective()

    print("= Effective light rig properties =")
    print("Type: " + str(threeDEffectiveData.light_rig.light_type))
    print("Direction: " + str(threeDEffectiveData.light_rig.direction))

#ExEnd:GetLightRigEffectiveData
